### DISCUSSION TITLE
Need help with nested floating element in webcomponent slot

### DISCUSSION THREAD

**Discussion by Spirielle** [floating-ui/floating-ui#3037] [Q&A] [answered]
Hi!
I'm trying to integrate Floating UI in my webcomponents.
Sadly, the placement is wrong when I try to nest floating elements. 

I believe it's because the offset parent of my floating element is not the same as the nearest positionned ancestor of the slot in which my floating element is slotted.

I made a codepen: https://codepen.io/Spirielle/pen/QWXzrwG

The DOM looks like this:
- Menu
	- shadowDom
		- Floating (positioned)
			- slot A
	- Popup (slotted in slot A)

I'm out of ideas, I hope someone can help 🙏

**Comment by atomiks** [comment, marked as answer]
https://floating-ui.com/docs/platform#shadow-dom-fix
Note: doesn't work with closed shadow roots, which isn't possible to fix

**Reply by Spirielle** [reply]
I thought I looked eveywhere 🤦‍♀️. Should've searched for "shadow dom" instead of "slot" and "webcomponent"
This helped me a lot, I truly appreciate, thanks 🙏
